File: test/shell-init.test.js
const { describe, it } = require("node:test");
const assert = require("node:assert/strict");

describe("shell-init", () => {
  it("outputs a wt() function definition", (t) => {
    let output = "";
    t.mock.method(console, "log", (s) => {
      output += s;
    });

    delete require.cache[require.resolve("../lib/commands/shell-init")];
    const shellInit = require("../lib/commands/shell-init");

    shellInit();

    assert.match(output, /wt\(\)/);
  });

  it("auto-cds after create and pr", (t) => {
    let output = "";
    t.mock.method(console, "log", (s) => {
      output += s;
    });

    delete require.cache[require.resolve("../lib/commands/shell-init")];
    const shellInit = require("../lib/commands/shell-init");

    shellInit();

    assert.match(output, /"create"/);
    assert.match(output, /"pr"/);
    assert.match(output, /command wt cd "\$name"/);
  });

  it("uses 'command wt' to avoid recursion", (t) => {
    let output = "";
    t.mock.method(console, "log", (s) => {
      output += s;
    });

    delete require.cache[require.resolve("../lib/commands/shell-init")];
    const shellInit = require("../lib/commands/shell-init");

    shellInit();

    assert.match(output, /command wt/);
  });
});

File: internal/cmd/shellinit_test.go
package cmd

import (
	"strings"
	"testing"
)

func TestShellInitCmd(t *testing.T) {
	t.Run("outputs a wt() function definition", func(t *testing.T) {
		d, stdout, _ := testDeps(nil, nil, nil)
		cmd := ShellInitCmd(d)
		cmd.SetArgs([]string{})
		cmd.Execute()

		if !strings.Contains(stdout.String(), "wt()") {
			t.Error("output should contain wt()")
		}
	})

	t.Run("auto-cds after create and pr", func(t *testing.T) {
		d, stdout, _ := testDeps(nil, nil, nil)
		cmd := ShellInitCmd(d)
		cmd.Execute()
		out := stdout.String()

		if !strings.Contains(out, `"create"`) {
			t.Error("should reference create")
		}
		if !strings.Contains(out, `"pr"`) {
			t.Error("should reference pr")
		}
		if !strings.Contains(out, `command wt cd "$name"`) {
			t.Error("should auto-cd after create/pr")
		}
	})

	t.Run("uses command wt to avoid recursion", func(t *testing.T) {
		d, stdout, _ := testDeps(nil, nil, nil)
		cmd := ShellInitCmd(d)
		cmd.Execute()

		if !strings.Contains(stdout.String(), "command wt") {
			t.Error("should use 'command wt' to avoid recursion")
		}
	})
}
